cpp/dolfinx/common/Timer.h

```diff
@@ -28,36 +28,40 @@ namespace dolfinx::common
 ///
 ///   list_timings();
 
+template <typename chrono_timer = std::chrono::high_resolution_clock>
 class Timer
 {
 public:
   /// Create timer
   ///
   /// If a task name is provided this enables logging to logger, otherwise (i.e.
   /// no task provided) nothing gets logged.
-  Timer(std::optional<std::string> task = std::nullopt);
+  Timer(std::optional<std::string> task = std::nullopt) : _task(task) {}
 
   /// Destructor
-  ~Timer();
+  ~Timer() = default;
 
   /// Zero and start timer
-  void start();
-
-  /// Resume timer. Not well-defined for logging timer
-  void resume();
```

Comment:
Any reason for removing this?

Reply:
Yes - the handling of the time points of `std::chrono` are not straight forward to extend to this case. We store the initial `start_time` and then compute the time when `stop` is called by the difference of current time to it. However if we would want to restart we need to store both an elapsed time and the new start point to maintain that property (or do something nasty and move the start time back by the already elapsed time). As it wasn't used anywhere I thought it would be easier to just remove it and not deal with that.